pkg/operator/ceph/cluster/nodedaemon/crash.go

```diff
@@ -49,6 +49,9 @@ func (r *ReconcileNode) createOrUpdateCephCrash(node corev1.Node, tolerations []
 		return controllerutil.OperationResultNone, errors.Errorf("label key %q does not exist on node %q", corev1.LabelHostname, node.GetName())
 	}
 	deploy := &appsv1.Deployment{
+
+		Spec: appsv1.DeploymentSpec{},
+
```

Comment:
I think this is unnecessary. I see that `deploy.Spec.Template` is already successfully being set in the `mutateFunc` code below.

Reply:
@BlaineEXE  wrote:
> I think this is unnecessary. I see that `deploy.Spec.Template` is already successfully being set in the `mutateFunc` code below.


yes, it sets `deploy.Spec.Template` but  I think that is problematic if `deploy.Spec` is not set.

Reply:
@BlaineEXE , you were of course right: The line is unnecessary. I removed it again.

Reply:
Does the code crash if this is removed?

Reply:
no, the latest state of the PR   with that code removed does not crash for me locally any more. As mentioned above, I updated the PR after verifying this locally.